spec/presenters/view_hook_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+
+RSpec.describe ViewHook do
+  let(:view_hook) { described_class.new }
+
+  describe "find" do
+    before do
+      view_hook.add_hook("view1", "placement1", "partial")
+      view_hook.add_hook("view1", "placement1", "partial2")
+    end
+
+    it "returns the correct partials" do
+      expect(view_hook.find("view1", "placement1")).to eq(["partial", "partial2"])
+    end
+
+    it "returns an empty array if the view is not found" do
+      expect(view_hook.find("view2", "placement1")).to eq([])
+    end
+
+    it "returns an empty array if the placement is not found" do
+      expect(view_hook.find("view1", "placement2")).to eq([])
+    end
+
+    it "finds using symbols" do
+      expect(view_hook.find(:view1, :placement1)).to eq(["partial", "partial2"])
+    end
+  end
+
+  describe "adding with symbols" do
+    before do
+      view_hook.add_hook(:view1, :placement, :partial)
+    end
+
+    it "finds by symbol" do
+      expect(view_hook.find(:view1, :placement)).to eq(["partial"])
+    end
+
+    it "finds by string" do
+      expect(view_hook.find("view1", "placement")).to eq(["partial"])
+    end
+  end
+
+  describe "render_view_hook" do
+    let(:context) { double }
+    before do
+      view_hook.add_hook("view1", "placement1", "partial")
+      view_hook.add_hook("view1", "placement1", "partial2")
```

Comment:
Maybe add a test to make sure the dots-to-underscore/slash translation is doing what we expect?